File: Eleicao-Java/TrabalhPROG3_2018_2/src/Candidato.java
package pkg_01;

import java.util.Comparator;

/**
 * Classe que implementa atributos e métodos de um Candidato em uma eleição.
 */
public class Candidato implements Comparator<Candidato> {
	private String nome; // Nome do candidato.
	private Partido partido; // Partido do candidato.
	private int votos_recebidos; // Identifica o numero de votos recebidos pelo candidato.
	private boolean eleito; // True significa que foi eleito.
							// False significa que nao foi eleito.

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public Partido getPartido() {
		return partido;
	}

	public void setPartido(Partido partido) {
		this.partido = partido;
	}

	public int getVotos_recebidos() {
		return votos_recebidos;
	}

	public void setVotos_recebidos(int votos_recebidos) {
		this.votos_recebidos = votos_recebidos;
	}

	public boolean isEleito() {
		return eleito;
	}

	public void setEleito(boolean eleito) {
		this.eleito = eleito;
	}

	@Override
	public int compare(Candidato a, Candidato b) {
		return a.getVotos_recebidos() - b.getVotos_recebidos();
	}

	@Override
	public String toString() {
		if (this.getPartido().getColigacao().getPartidos().size() > 1) {
			return this.getNome() + " (" + this.getPartido().getNome() + ", " + this.getVotos_recebidos() + " votos)"
					+ " - " + this.getPartido().getColigacao();
		} else {
			return this.getNome() + " (" + this.getPartido().getNome() + ", " + this.getVotos_recebidos() + " votos)";
		}
	}

}

File: Eleicao-Cplusplus/Candidato.cpp
#include "Candidato.h"

Candidato::Candidato() {
	this->votos_recebidos = 0;
	this->eleito = false;
}

Candidato::~Candidato() {
}

string Candidato::getNome() const {
	return nome;
}

void Candidato::setNome(const string& nome) {
	this->nome = nome;
}

Partido* Candidato::getPartido() const{
	return this->partido;
}

void Candidato::setPartido(Partido& partido){
	this->partido = &partido;
}

int Candidato::getVotos_recebidos() const{
	return votos_recebidos;
}

void Candidato::setVotos_recebidos(const int& votos_recebidos){
	this->votos_recebidos = votos_recebidos;
}

bool Candidato::isEleito() const{
	return eleito;
}

void Candidato::setEleito(bool eleito){
	this->eleito = eleito;
}

void Candidato::toString() const{
	if (this->getPartido()->getColigacao()->numDePartidos() > 1) {
		cout << this->getNome() << " (" << this->getPartido()->getNome() << ", ";
		cout << this->getVotos_recebidos() << " votos) - ";
		cout << this->getPartido()->getColigacao()->toString();

	} else {
		cout << this->getNome() << " (" << this->getPartido()->getNome() << ", ";
		cout << this->getVotos_recebidos() << " votos)";
	}
}

bool comparaCandidatos (Candidato*& a, Candidato*& b){
	return a->getVotos_recebidos() > b->getVotos_recebidos();
}